src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,495 @@
+---
+title: WAL Failover
+summary: Mitigate the effects of disk stalls in cloud deployments by failing over WAL writes to a secondary store.
+toc: true
+docs_area: deploy
+---
+
+{% include {{ page.version.version }}/wal-failover-intro.md %}
+
+This page has detailed information about WAL failover, including:
+
+- How WAL failover works.
+- How to enable WAL failover.
+- How to disable WAL failover.
+- How to test WAL failover.
+- How to monitor WAL failover.
+- How to configure WAL failover in multi-store configurations.
+- Frequently Asked Questions about WAL failover.
+
+For basic information about WAL failover, see [`cockroach start` > WAL failover]({% link {{ page.version.version }}/cockroach-start.md %}#write-ahead-log-wal-failover).
+
+## Why WAL Failover?
+
+In cloud environments transient [disk stalls]({% link {{ page.version.version }}/cluster-setup-troubleshooting.md %}#disk-stalls) are common, often lasting on the order of several seconds. This will negatively impact latency for the user-facing foreground workload. In the field, we have observed that stalls occur most frequently while writing to the WAL. While we cannot prevent disk stalls, we can minimize their impact on CockroachDB. That is where WAL failover comes into play.
```

Comment:
> In the field, we have observed that stalls occur most frequently while writing to the WAL

This sentence isn't quite right. Stalls apply to the various files of the storage engine with equal frequency, but a stall of the write-ahead log is the most impactful to foreground latencies. (Because most other writes, like those of flushes and compactions happen asynchronously in the background and foreground operations do not need to wait for them.)

Reply:
Updated to the following based on your comment - PTAL

> In the field, we have observed that stalls while writing to the WAL are the most impactful to foreground latencies. Most other writes, such as flushes and compactions, happen asynchronously in the background and foreground operations do not need to wait for them.